File: REST_module/src/main/java/com/rzk/nurdor/security/NetworkUtils.java
package com.rzk.nurdor.security;

import java.net.DatagramSocket;
import java.net.InetAddress;
import java.net.SocketException;
import java.net.UnknownHostException;

public class NetworkUtils {

    public static String getIPV4() {
        try (final DatagramSocket datagramSocket = new DatagramSocket()) {
            datagramSocket.connect(InetAddress.getByName("8.8.8.8"), 12345);
            return datagramSocket.getLocalAddress().getHostAddress();
        } catch (SocketException | UnknownHostException e) {
            throw new RuntimeException(e);
        }
    }

    public static void main(final String[] args) {
        try {
            System.out.println(NetworkUtils.getIPV4());
        } catch (Exception e) {
            e.fillInStackTrace();
        }
    }
}

File: RMA_Project_v2/app/src/main/java/com/example/rma_nurdor_project_v2/utils/NetworkUtils.kt
package com.example.rma_nurdor_project_v2.utils

import java.net.DatagramSocket
import java.net.InetAddress

object NetworkUtils {
    fun getIPV4(): String? {
        return DatagramSocket().use { socket ->
            socket.connect(InetAddress.getByName("8.8.8.8"), 12345)
            socket.localAddress.hostAddress
        }
    }
}